src/categorical_algebra/FinCats.jl

```diff
@@ -104,17 +103,81 @@ is_discrete(C::FinCat) = isempty(hom_generators(C))
 """
 is_free(C::FinCat) = isempty(equations(C))
 
+"""Graph underlying a finitely presented category whose 
+object and
+hom generators are indexable, other
+than one explicitly generated by a graph.
+"""
+function graph(C::FinCat;maps::Bool=false,inv::Bool=false,named::Bool=false) 
```

Comment:
Rather than having all these keyword arguments, can we just return a named graph whose names are the object/hom *generators*? Then lookup and indexing will give you both the forward and backward mappings.

This does suggest that maybe thing being returned should not be called "named graph".

Reply:
Hmmmmmmm, I'm pretty sure I got this working without returning maps at all but leaving the names as symbols.